app/assets/javascripts/pretty-text/engines/discourse-markdown/quotes.js.es6

```diff
@@ -34,9 +34,30 @@ const rule = {
       }
     }
 
-
-    let token    = state.push('bbcode_open', 'aside', 1);
-    token.attrs  = [['class', 'quote']];
+    if (options.lookupAvatarByPostNumber) {
+      // client-side, we can retrieve the avatar from the post
+      avatarImg = options.lookupAvatarByPostNumber(postNumber, topicId);
+    } else if (options.lookupAvatar) {
+      // server-side, we need to lookup the avatar from the username
+      avatarImg = options.lookupAvatar(username);
+    }
+    
+    if (options.lookupPrimaryUserGroupByPostNumber) {
+      // client-side, we can retrieve the primary user group from the post
+      primaryGroupName = options.lookupPrimaryUserGroupByPostNumber(postNumber, topicId);
+    } else if (options.lookupPrimaryUserGroup) {
+      // server-side, we need to lookup the primary user group from the username
+      primaryGroupName = options.lookupPrimaryUserGroup(username);
+    }
+    
+    let token   = state.push('bbcode_open', 'aside', 1);
+    token.attrs = [];
+    
+    if (primaryGroupName) {
```

Comment:
You are returning empty strings for this function - so that would pass the empty check and allow this to end up as `group-` sometimes. You should check for length.